eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/responses/m2w/M2WebEwonEBDInstantValuesResponse.java

```diff
@@ -53,21 +53,31 @@ public List<M2WebEwonEBDInstantValue> getInstantValues() {
    * @param responseString The response string to parse.
    * @return An instance of {@link M2WebEwonEBDInstantValuesResponse} parsed from the response
    *     string.
-   * @throws IllegalStateException If a necessary parameter was not specified. Currently this means
+   * @throws IllegalStateException If a necessary parameter was not specified. Currently, this means
    *     that both the mapping strategy and the bean type are not set, so it is impossible to
    *     determine a mapping strategy.
    * @since 1.0.0
    */
   public static M2WebEwonEBDInstantValuesResponse getFromString(String responseString)
       throws IllegalStateException {
     // Parse CSV response into list of tag objects
-    List<M2WebEwonEBDInstantValue> instantValues =
-        new CsvToBeanBuilder<M2WebEwonEBDInstantValue>(new StringReader(responseString))
-            .withSeparator(EWON_EBD_INSTANTVALUES_SEPARATOR_CHAR)
-            .withType(M2WebEwonEBDInstantValue.class)
-            .build()
-            .parse();
+    M2WebEwonEBDInstantValuesResponse response;
+    try {
+      List<M2WebEwonEBDInstantValue> instantValues =
+          new CsvToBeanBuilder<M2WebEwonEBDInstantValue>(new StringReader(responseString))
+              .withSeparator(EWON_EBD_INSTANTVALUES_SEPARATOR_CHAR)
+              .withType(M2WebEwonEBDInstantValue.class)
+              .build()
+              .parse();
+      response = new M2WebEwonEBDInstantValuesResponse(instantValues);
+    } catch (Exception e1) {
+      try {
+        response = getFromJson(responseString, M2WebEwonEBDInstantValuesResponse.class);
```

Comment:
getFromJson does not return M2WebEwonEBDInstantValuesResponse. Does this matter?
not sure what response.getInstantValues() would do

Reply:
```java
  /**
   * Gets an instance of {@link M2WebEwonEBDResponse} from a JSON string.
   *
   * @param json The JSON string to parse.
   * @return An instance of {@link M2WebEwonEBDResponse} parsed from the JSON string.
   * @throws JsonSyntaxException if the JSON string is not valid or does not match the {@link
   *     M2WebEwonEBDResponse} class.
   * @since 1.0.0
   */
  public static <T extends M2WebEwonEBDResponse> T getFromJson(String json, Class<T> clazz)
      throws JsonSyntaxException {
    return new Gson().fromJson(json, clazz);
  }
```
Both Talk2MResponse#getFromJson(String,Class<T>) and in this case, M2WebEwonEBDResponse#getFromJson(String,Class<T>) utilize generics/type parameters that match the upstream call to `new Gson().fromJson(json,clazz);`. The `Class<t>` passed as the second parameter defines the return type, and the Java environment enforces the return type (i.e., you can't pass one class as the second parameter, then store the return value as a different class/type.)

Simplified, the `<T extends M2WebEwonEBDResponse>` defines the type parameter (in this case, one that must extend M2WebEwonEBDResponse). 
-> Then, the method return type is specified as `T`, and the second parameter is a class that must be of type `T` (`<T>`). 
-> Typically with Java, you only have to specify the type parameter once, and the rest is inferred by the JVM. In this case, we require passing a class as the second parameter, which defines the type parameter.